### ISSUE TITLE
EcsOnRemove Segfault on Component Access with Matched Container Entities

### ISSUE THREAD

**Issue by teksoc-1** [SanderMertens/flecs#171]
**Describe the bug**
A segfault occurs when accessing the component received on an EcsOnRemove system in the case where a matched entity is a container.

**To Reproduce**
```C
#include "flecs.h"
#include <stdio.h>

#define TRIGGER_CRASH 1

typedef int Parent;
typedef int Car;
typedef int Child;

void
OnRemoveCar( ecs_rows_t * rows )
{
    ECS_COLUMN( rows, Car, cars, 1 );
    
    for ( unsigned int i = 0; i < rows->count; i++ )
    {
        // if TRIGGER_CRASH, triggers segfault
        printf( "cars[ %i ]: %i\n", i, cars[ i ] );
    }
}

int 
main( int argc, const char * argv[] )
{
    // Create the world
    ecs_world_t * world = ecs_init();
    
    // Register components
    ECS_COMPONENT( world, Parent );
    ECS_COMPONENT( world, Car );
    ECS_COMPONENT( world, Child);
    
    // Register a reactive system
    ECS_SYSTEM( world, OnRemoveCar, EcsOnRemove, Car );
    
    // Create a parent and give them a car
    ecs_entity_t parent = ecs_new( world, Parent );
    ecs_add( world, parent, Car );
    
#if TRIGGER_CRASH
    // Create a child and have it adopted by the parent
    ecs_entity_t child = ecs_new( world, Child );
    ecs_adopt( world, child, parent );
#endif
    
    // Delete the parent, which triggers the reactive on remove system
    ecs_delete( world, parent );
    
    // We're done
    ecs_fini( world );

    return 0;
}
```

**Expected behavior**
Setting TRIGGER_CRASH to 0 demonstrates expected behavior: successful execution and termination of program yielding below output:
```
cars[ 0 ]: <some integer value>
```
**Additional context**
Accessing `rows->entities[ i ]` inside of `OnRemoveCar` also triggers segfault, which seems fundamentally to be the issue at hand

**Comment by teksoc-1** [comment]
Found that if one manually calls `_ecs_remove()` prior to the `ecs_delete()` the `OnRemoveCar()` works

**Comment by SanderMertens** [comment]
Fixed the issue. Parent entities are "watched" entities which internally means that Flecs flips the sign bit on the row in the entity index. The remove used a code path that corrected for this, but the delete code path is slightly different, and this caused a negative row to end up being used to calculate the component pointer.

This issue was already fixed on bleeding_edge, I just pushed the fix to master as well.
